src/Api/DateTimeResult.php

```diff
@@ -31,12 +30,39 @@ public static function fromEpoch($unixTimestamp)
      *
      * @return DateTimeResult
      */
-    public static function fromISO8601($iso8601Timestamp)
-    {
+    public static function fromISO8601($iso8601Timestamp) {
         if (is_numeric($iso8601Timestamp) || !is_string($iso8601Timestamp)) {
             throw new ParserException('Invalid timestamp value passed to DateTimeResult::fromISO8601');
         }
-        return new DateTimeResult($iso8601Timestamp);
+        $timeIso8601 = self::toPrecisionMs($iso8601Timestamp);
+        return new DateTimeResult($timeIso8601);
+    }
+
+    /**
+     * 
+     * @param string $iso8601Timestamp
+     * 
+     * @return string
+     */
+    private static function toPrecisionMs($iso8601Timestamp) {
+        $timeIso8601 = strtoupper($iso8601Timestamp);
+        $startPos = strpos($timeIso8601, '.');
+        if ($startPos != false) {
+            $startPos += 1;
+            $endPos = strpos($timeIso8601, '+');
+            if ($endPos == false) {
```

Comment:
This part could be fewer lines:
if ($endPos = strpos($timeIso8601, '+') == false) {
  if ($endPos = strpos($timeIso8601, "Z") == false) {
  $endPos = strlen($timeIso8601);
  }
}